### PR TITLE
Implement te-audio

### PR DESCRIPTION
**Pull request by zcuric**
### This PR
- [x] Implements audio teaching element

### CODE REVIEW

File: src/teaching-element/Audio.vue [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Aplayer.disableVersionBadge = true;
+
+const defaultAudioOptions = {
+  title: 'Audio track',
+  artist: ' ',
+  pic: ' '
+};
+
+export default {
+  name: 'te-file',
```
Comment (abasic) [inline src/teaching-element/Audio.vue]:
```suggestion
  name: 'te-audio',
```

File: src/teaching-element/Audio.vue [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    <aplayer
+      v-if="url"
+      v-bind="playerOptions"
+      :music="music" />
+  </div>
+</template>
+
+<script>
+import Aplayer from 'vue-aplayer';
+Aplayer.disableVersionBadge = true;
```
Comment (abasic) [inline src/teaching-element/Audio.vue]: Missing empty line above.

File: src/teaching-element/Audio.vue [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+const defaultAudioOptions = {
```
Comment (abasic) [inline src/teaching-element/Audio.vue]: Maybe this should be a function that returns an object.

Comment (zcuric) [reply src/teaching-element/Audio.vue]: Any particular reason why?

File: src/teaching-element/Audio.vue (hunk 2 above)
Comment (zcuric) [reply src/teaching-element/Audio.vue]: 👍

File: src/teaching-element/Audio.vue (hunk 1 above)
Comment (zcuric) [reply src/teaching-element/Audio.vue]: 👍

File: src/teaching-element/Audio.vue (hunk 3 above)
Comment (zcuric) [reply src/teaching-element/Audio.vue]: Nevermind.

File: src/teaching-element/Audio.vue [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  pic: ' '
+});
+
+export default {
+  name: 'te-audio',
+  inheritAttrs: false,
+  props: {
+    url: { type: String, required: true },
+    audioOptions: { type: Object, default: defaultAudioOptions },
+    playerOptions: { type: Object, default: () => ({}) }
```
Comment (kronicker) [inline src/teaching-element/Audio.vue]: `tailor-teaching-elements` is one composite component and is consumed as such which means that all of the props (coming from outside) are shared and you have the same prop as `te-video` (`playerOptions`) which makes it hard for the consumer to maintain them.

File: src/teaching-element/Audio.vue [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  <div class="te-audio">
+    <aplayer
+      v-if="url"
+      v-bind="playerOptions"
+      :music="music" />
+  </div>
+</template>
+
+<script>
+import Aplayer from 'vue-aplayer';
```
Comment (kronicker) [inline src/teaching-element/Audio.vue]: Use [multi-word component names](https://vuejs.org/v2/style-guide/#Multi-word-component-names-essential)

File: src/teaching-element/Audio.vue [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+export default {
+  name: 'te-audio',
+  inheritAttrs: false,
+  props: {
+    url: { type: String, required: true },
+    audioOptions: { type: Object, default: defaultAudioOptions },
+    playerOptions: { type: Object, default: () => ({}) }
+  },
+  computed: {
+    music() {
```
Comment (kronicker) [inline src/teaching-element/Audio.vue]: `audio`

Comment (kronicker) [reply src/teaching-element/Audio.vue]: Also can be inline lambda

File: src/teaching-element/Audio.vue (hunk 4 above)
Comment (kronicker) [reply src/teaching-element/Audio.vue]: Come to think about it, maybe we should make `playerOptions` and `audioOptions` as one prop.

Comment (zcuric) [reply src/teaching-element/Audio.vue]: > `tailor-teaching-elements` is one composite component and is consumed as such which means that all of the props (coming from outside) are shared and you have the same prop as `te-video` (`playerOptions`) which makes it hard for the consumer to maintain them.

I don't understand what is your point here.  Consumer doesn't maintain anything he passes down the props which are being used. He can name data for that prop what ever he want and pass it down to `te-audio` or `te-video` as `playerOptions` which are isolated from each other. 

Let's not complicate things.

Comment (zcuric) [reply src/teaching-element/Audio.vue]: > Come to think about it, maybe we should make `playerOptions` and `audioOptions` as one prop.

There not the same type of options and they should stay separated. `audioOptions` are options for audio files (title, pic, artist), `playerOptions` are player options. Check [the docs](https://github.com/SevenOutman/vue-aplayer/blob/develop/docs/README.md).

Comment (zcuric) [reply src/teaching-element/Audio.vue]: > `tailor-teaching-elements` is one composite component and is consumed as such which means that all of the props (coming from outside) are shared and you have the same prop as `te-video` (`playerOptions`) which makes it hard for the consumer to maintain them.

I'll name it `audioPlayerOptions`.

**Pull request merged**